presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -581,6 +640,32 @@ public InternalResourceGroup getOrCreateSubGroup(String name)
         }
     }
 
+    public int getRunningTaskCount()
+    {
+        if (subGroups().isEmpty()) {
+            return runningQueries.stream()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.



**Problem**  
 This line may not be thread-safe: **646**. At this line the following field is not protected by synchronization: `runningQueries`. Your code typically protects this field with synchronization on object: `root`.

**Fix**  
 Consider protecting the above field with synchronization on object: `root`. For example, like at lines: **316** and **934**.

**More info**  
 The following lines are where the above field is already protected by synchronization on the above object: **279**, **316**, **620**, **752**, **774**, **786**, **787**, **813**, **923**, **934**, and **991**.